service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/AbstractControllerTest.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import static org.apache.http.HttpStatus.SC_OK;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+abstract class AbstractControllerTest {
+
+    private String eventName;
+
+    final String userAuthToken = "Bearer token";
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @Autowired
+    ObjectMapper mapper;
+
+    AbstractControllerTest(String eventName) {
+        this.eventName = eventName;
+    }
+
+    AboutToStartOrSubmitCallbackResponse postAboutToStartEvent(CaseDetails caseDetails,
+                                                               int expectedStatus) {
+        return postEvent(String.format("/callback/%s/about-to-start", eventName), caseDetails, expectedStatus);
+    }
+
+    AboutToStartOrSubmitCallbackResponse postAboutToStartEvent(CaseDetails caseDetails) {
+        return postAboutToStartEvent(caseDetails, SC_OK);
+    }
+
+    AboutToStartOrSubmitCallbackResponse postMidEvent(CaseDetails caseDetails, int expectedStatus) {
```

Comment:
Not for this implementation as nothing currently does it but I think that I have heard that there might be situations in the future where there might be more that one mid event per class, so an extra part bit may need to be added here to accommodate for that, assume the url is going to be the format of `/callback/eventName-midEventName/mid-event` but this can be handled in the future if we actually do that

Reply:
as discussed, it is not part of this PR